src/theory/quantifiers/ceg_t_instantiator.cpp

```diff
@@ -1038,60 +1041,79 @@ Node BvInstantiator::hasProcessAssertion(CegInstantiator* ci,
     {
       return lit;
     }
-  } else {
+  }
+  else
+  {
     bool useSlack = false;
-    if (k == EQUAL && atom[0].getType().isBitVector())
+    if (k != EQUAL && k != BITVECTOR_ULT && k != BITVECTOR_SLT)
+    {
+      // others are unhandled
+      return Node::null();
+    }
+    else if (!atom[0].getType().isBitVector())
+    {
+      return Node::null();
+    }
+
+    NodeManager* nm = NodeManager::currentNM();
+    Node s = atom[0];
+    Node t = atom[1];
+
+    Node sm = ci->getModelValue(atom[0]);
+    Node tm = ci->getModelValue(atom[1]);
+    Assert(!sm.isNull() && sm.isConst());
+    Assert(!tm.isNull() && tm.isConst());
+    Trace("cegqi-bv") << "Model value: " << std::endl;
+    Trace("cegqi-bv") << "   " << s << " " << k << " " << t << " is "
+                      << std::endl;
+    Trace("cegqi-bv") << "   " << sm << " <> " << tm << std::endl;
+
+    if (options::cbqiBvIneqMode() == CBQI_BV_INEQ_EQ_SLACK)
     {
-      // always use slack for disequalities
       useSlack = true;
-    } else if (k == BITVECTOR_ULT || k == BITVECTOR_SLT) {
-      if (options::cbqiBvIneqMode() == CBQI_BV_INEQ_EQ_SLACK)
+    }
+    else if (k == EQUAL)
+    {
+      Node comp = nm->mkNode(BITVECTOR_ULT, sm, tm);
+      comp = Rewriter::rewrite(comp);
```

Comment:
Why not use the `BitVector` values of `sm` and `tm` for comparison? Then you wouldn't have to invoke the rewriter, right?

Reply:
You could, but I prefer this way since it is safer.  It is theoretically possible that sm and tm won't be constants, there is an assertion that checks this but I'd prefer not crashing in production if this were to occur.